lib/spack/spack/directory_layout.py

```diff
@@ -260,9 +260,10 @@ def check_installed(self, spec):
             return None
 
         if not os.path.isfile(spec_file_path):
-            raise InconsistentInstallDirectoryError(
-                'Install prefix exists but contains no spec.yaml:',
-                "  " + path)
+            return None
+#            raise InconsistentInstallDirectoryError(
+#                'Install prefix exists but contains no spec.yaml:',
+#                "  " + path)
```

Comment:
Is this intended to be part of this PR? This might be a more controversial change.